src/net/benmann/evald/Evald.java

```diff
@@ -193,27 +282,39 @@ private int addVariableToList(String token, Double value) {
 
         Integer result = keyIndexMap.get(token);
         if (result == null) {
-            result = valueList.size();
+            result = valueArraySize++;
             keyIndexMap.put(token, result);
-            valueList.add(value);
-
-            if (valueArray != null && valueArray.length <= result) {
-                double[] oldValues = valueArray;
-                valueArray = new double[result + ARRAY_RESIZE_BUFFER];
-                System.arraycopy(oldValues, 0, valueArray, 0, oldValues.length);
-                for (SetValueArrayCallback callback : valueArrayCallbacks) {
-                    callback.setValueArray(valueArray);
-                }
-            }
-        } else {
-        	valueList.set(result, value);
+            validateValueArrayIndex(result);
+            variableToken[result] = token;
         }
+        valueArray[result] = value;
```

Comment:
Is this necessary/correct with the if below?

Reply:
Good question! The initialisation of `valueArray` has changed, so I suspect the latter (with its null check) can go.